tools/code-owners-parser/Azure.Sdk.Tools.RetrieveCodeOwners/Program.cs

```diff
@@ -125,5 +119,26 @@ private static bool VerifyAlias(string uriStub)
                 throw;
             }
         }
+
+        private static CodeOwnerEntry findClosestMatch(string filePath, List<CodeOwnerEntry> entries)
```

Comment:
We should put this helper in the code owners parsing library. 

Also for .NET you should start the function with a capital letter so `FindClosestMatch`.

Reply:
I would move this to the CodeOwnersFile class and call it `ParseAndFindClosestMatch(string codeownersfile, string target path)`  and have it return a list of contacts.

Reply:
The class is setting PRLabels and ServiceLabels as well. Contact itself will lose the info

Reply:
Perharps you can use this helper for 
https://github.com/Azure/azure-sdk-tools/blob/bd8075587ee3a4370bfcd7e0a1508c44ad128c1f/tools/notification-configuration/github-codeowner-subscriber/Program.cs#L135

Reply:
The program here should be fairly simple and most of the selection and parsing logic for both codeowners and github alias processing should be inside of the codeowners library.

Reply:
>The class is setting PRLabels and ServiceLabels as well. Contact itself will lose the info

We don't have to worry about those really those are on the entries but we aren't really even going to return entries from this helper it will just return a list of github alias or perhaps a set of user vs team alias.